crates/op-rbuilder/src/primitives/reth/engine_api_builder.rs

```diff
@@ -0,0 +1,350 @@
+            name: OP_NAME_CLIENT.to_string(),
+            version: CARGO_PKG_VERSION.to_string(),
+            commit: VERGEN_GIT_SHA.to_string(),
+        };
+        let inner = reth_rpc_engine_api::EngineApi::new(
+            ctx.node.provider().clone(),
+            ctx.config.chain.clone(),
+            ctx.beacon_engine_handle.clone(),
+            PayloadStore::new(ctx.node.payload_builder_handle().clone()),
+            ctx.node.pool().clone(),
+            Box::new(ctx.node.task_executor().clone()),
+            client,
+            EngineCapabilities::new(OP_ENGINE_CAPABILITIES.iter().copied()),
+            engine_validator,
+            ctx.config.engine.accept_execution_requests_hash,
+        );
+
+        Ok(OpEngineApiExt::new(
+            OpEngineApi::new(inner),
+            engine_peers,
+            ctx.node.task_executor().clone(),
+        ))
+    }
+}
+
+pub struct OpEngineApiExt<Provider, EngineT: EngineTypes, Pool, Validator, ChainSpec> {
+    inner: OpEngineApi<Provider, EngineT, Pool, Validator, ChainSpec>,
+    engine_peers: Vec<EnginePeer>,
+    task_executor: TaskExecutor,
+}
+
+impl<Provider, EngineT, Pool, Validator, ChainSpec>
+    OpEngineApiExt<Provider, EngineT, Pool, Validator, ChainSpec>
+where
+    Provider: HeaderProvider + BlockReader + StateProviderFactory + 'static,
+    EngineT: EngineTypes<ExecutionData = OpExecutionData>,
+    Pool: TransactionPool + 'static,
+    Validator: EngineValidator<EngineT>,
+    ChainSpec: EthereumHardforks + Send + Sync + 'static,
+{
+    pub fn new(
+        engine: OpEngineApi<Provider, EngineT, Pool, Validator, ChainSpec>,
+        engine_peers: Vec<EnginePeer>,
+        task_executor: TaskExecutor,
+    ) -> Self {
+        Self {
+            inner: engine,
+            engine_peers,
+            task_executor,
+        }
+    }
+
+    /// Multiplexes the given engine API call to all configured peers
+    async fn multiplex_fork_choice_update(
+        &self,
+        method: &str,
+        fork_choice_state: &ForkchoiceState,
+        payload_attributes: &Option<EngineT::PayloadAttributes>,
+    ) {
+        if self.engine_peers.is_empty() {
+            return;
+        }
+
+        let task_executor = &self.task_executor;
+
+        for peer in &self.engine_peers {
+            let client = peer.http_client();
+            let method = method.to_string();
+            let fork_choice_state = *fork_choice_state;
+            let payload_attributes = payload_attributes.clone();
+            task_executor.spawn(Box::pin(async move {
+                // For now, just log the multiplexing until we implement proper Alloy provider calls
+                let res: Result<ForkchoiceUpdated, jsonrpsee_core::client::error::Error> = client
+                    .request(method.as_str(), (fork_choice_state, payload_attributes))
+                    .await;
+                if let Err(err) = res {
+                    warn!("Failed to multiplex FCU, error: {err}");
```

Comment:
I would also include the peer info that we've failed to multiplex to in the log warn message.